hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/handlers/EthereumTransactionHandler.java

```diff
@@ -64,41 +72,55 @@ public EthereumTransactionHandler(
     }
 
     @Override
-    public void preHandle(@NonNull final PreHandleContext context) {
+    public void preHandle(@NonNull final PreHandleContext context) throws PreCheckException {
         requireNonNull(context);
         final var body = context.body().ethereumTransactionOrThrow();
         final var fileStore = context.createStore(ReadableFileStore.class);
         final var hederaConfig = context.configuration().getConfigData(HederaConfig.class);
-        final var ethTxData = callDataHydration
-                .tryToHydrate(body, fileStore, hederaConfig.firstUserEntity())
-                .ethTxData();
-        if (ethTxData != null) {
-            // Ignore the return value; we just want to cache the signature for use in handle()
-            ethereumSignatures.computeIfAbsent(ethTxData);
-        }
+        final var hydratedTx = callDataHydration.tryToHydrate(body, fileStore, hederaConfig.firstUserEntity());
+
+        validateTruePreCheck(hydratedTx.status() == OK, hydratedTx.status());
+
+        final var ethTxData = hydratedTx.ethTxData();
+        validateTruePreCheck(ethTxData != null, INVALID_ETHEREUM_TRANSACTION);
+
+        // Ignore the return value; we just want to cache the signature for use in handle()
+        ethereumSignatures.computeIfAbsent(ethTxData);
     }
 
     @Override
     public void handle(@NonNull final HandleContext context) throws HandleException {
         // Create the transaction-scoped component
         final var component = provider.get().create(context, ETHEREUM_TRANSACTION);
 
-        // Run its in-scope transaction and get the outcome
-        final var outcome = component.contextTransactionProcessor().call();
+        final var hevmTransactionFactory = component.contextTransactionProcessor().hevmTransactionFactory;
+        final var hevmTransaction = hevmTransactionFactory.fromHapiTransaction(context.body());
+
+        final var accountStore = context.readableStore(ReadableAccountStore.class);
+        final var sender = accountStore.getAccountById(Objects.requireNonNull(hevmTransaction.senderId()));
 
         // Assemble the appropriate top-level record for the result
         final var ethTxData =
                 requireNonNull(requireNonNull(component.hydratedEthTxData()).ethTxData());
+
+        // Run its in-scope transaction and get the outcome
+        final var outcome = component.contextTransactionProcessor().call();
+
         final var recordBuilder = context.recordBuilder(EthereumTransactionRecordBuilder.class)
                 .ethereumHash(Bytes.wrap(ethTxData.getEthereumHash()))
-                .status(outcome.status());
+                .status(outcome.status())
+                .feeChargedToPayer(outcome.tinybarGasCost());
```

Comment:
I think the two changes from [this commit](https://github.com/hashgraph/hedera-services/commit/1313d2ead9c91d5d79cbb6677eca632a2ffe16d7) that re-use the existing `recordBuilder.withTinybarGasFee()` method is preferable. 

Very nice catch though. 🙌